test/jdk/java/awt/im/PinyinIMCommaTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import jtreg.SkippedException;
+import sun.awt.OSInfo;
+
+/*
+ * @test
+ * @bug 8148984
+ * @summary Verify that Chinese comma can be entered in JTextField
+ *          with Pinyin input method (IM) on OS X.
+ * @requires (os.family == "mac")
+ * @modules java.desktop/sun.awt
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @run main/manual PinyinIMCommaTest
+ */
+
+public class PinyinIMCommaTest {
+    private static final String INSTRUCTIONS = """
```

Comment:
`INSTRUCTIONS ` can be moved inside main  (Other test also)

Reply:
Since it is okay to have it either as static var or within main, I'm going to leave it as it is now.